File: frontend/src/schema/transaction.ts
import { z } from 'zod';

export const TransactionSchema = z.object({
  id: z.string(),
  buyerId: z.string(),
  sellerId: z.string(),
  vehicleListingId: z.string(),
  amount: z.number(),
  status: z.string(),
  stripePaymentIntentId: z.string(),
  createdAt: z.date(),
  updatedAt: z.date(),
});

export type Transaction = z.infer<typeof TransactionSchema>;

File: backend/app/schema/transaction.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class Transaction(BaseModel):
    id: str
    buyer_id: str
    seller_id: str
    vehicle_listing_id: str
    amount: float
    status: str
    stripe_payment_intent_id: str
    created_at: datetime
    updated_at: datetime